README.md

```diff
@@ -4,3 +4,190 @@ lua-client
 [![Build Status](https://travis-ci.org/neovim/lua-client.svg?branch=master)](https://travis-ci.org/neovim/lua-client)
 
 Lua client for Neovim
+
+
+Basic usage
+-----------
+
+The module `nvim.nvim` provides a set of functions to connect to an instance of neovim.
+
+### Create an instance of the client
+
+First of all we need to get an instance of the client ready to talk with neovim.
+
+#### new_from_socket_file
+
+Connects into a running neovim instance using a socket file stream.
+
+```lua
+local Nvim = require('nvim.nvim')
+local nvim = Nvim.new_from_socket_file('/tmp/.nvim.socket')
+```
+
+#### new_from_socket
+
+Connects into a running neovim instance using a socket tcp stream.
+
+```lua
+local Nvim = require('nvim.nvim')
+local nvim = Nvim.new_from_socket('127.0.0.1', 6666)
+```
+
+#### new_from_process
+
+```lua
+local Nvim = require('nvim.nvim')
+local nvim = Nvim.new_from_process()
+```
+
+This code will create a new embedded neovim instance and will connect into it.
+
+
+### Using the client
+
+#### Listing the types available in the client
+
+Once we have a client we can list the types supported by neovim, the list of types include the basic types and the extended types.
+
+```lua
+local Nvim = require('nvim.nvim')
+local nvim = Nvim.new_from_process()
+
+for _, m in pairs(nvim._types) do
+  print(m)
+end
+```
+
+The output should be something like this:
+
+```
+Buffer[nvim ext type]
+Integer[nvim type]
+ArrayOf[nvim type]
+Window[nvim ext type]
+Dictionary[nvim type]
+Object[nvim type]
+Nil[nvim type]
+Tabpage[nvim ext type]
+Float[nvim type]
+void[nvim type]
+Array[nvim type]
+Boolean[nvim type]
+String[nvim type]
+```
+
+##### Basic types vs extended types
+
+Basic types are coerced into lua types directly while extended types implement a class of that type (Tabpage, Window and Buffer in the example). This means that wherever we have an extended type we need to use `:` to call any of its methods.
```

Comment:
Basic types are coerced to lua types directly whereas extended types implement their respective class (Tabpage, Window, and Buffer in the example). This means that we need to use `:` to call the methods of an extended type.